web-app/src/components/BookingModal/index.js

```diff
@@ -1,102 +1,160 @@
 import React from 'react';
-import DatePicker from 'react-datepicker';
 import { PropTypes as PT } from 'prop-types';
-import { Button, Modal } from '../common';
+import { Modal, Form, Input } from '../common';
 import FontAwesomeIcon from '@fortawesome/react-fontawesome';
 import { faTimes } from '@fortawesome/fontawesome-free-solid';
-import { StyleContainer, BookingStatus, BookingInputs, StatusDot, ButtonWrap } from './styled';
+import {
+  StyleContainer,
+  BookingStatus,
+  StatusDot,
+  FormContainer,
+} from './styled';
 import { statusText } from '../../utilities/holidayStatus';
 
 const BookingModal = props => {
   const {
     closeModal,
     showModal,
     booking,
-    changeStart,
-    changeEnd,
-    changeHalfday,
-    requestHoliday,
-    cancelHoliday,
-    updateHoliday,
-    changeWFH,
+    submitForm,
+    formStatus,
+    formIsValid,
   } = props;
 
-  const getActions = () => (
-    booking.title ?
-      <ButtonWrap>
-        <Button id="updateHolidayBtn" onClick={updateHoliday} label={'Update'} />
-        <Button id="cancelHolidayBtn" onClick={cancelHoliday} label={'Cancel Booking'} />
-      </ButtonWrap>
-      :
-      <Button id="requestHolidayBtn" onClick={requestHoliday} label={booking.isWFH ? 'Add' : 'Request'} />
-  );
+  console.log('formIsValid :', formIsValid);
+
+  const isDateDisabled = () => {
```

Comment:
You guys know the system better than I do but is this functionality what you want? Date is disabled if they're working from home or having a half day but if the booking duration is 0.5 then it should be false. Is 0.5 not a half day or is this expected behaviour?

Reply:
Yes, if requesting to work from home or take a half day, only one day can be requested, therefore only a start date is required. Hence end date is changed to equal start date, disabled and hidden from the UI